src/time/format/formatter.cr

```diff
@@ -139,51 +139,55 @@ struct Time::Format
       io << time.epoch
     end
 
-    def time_zone
-      case time.kind
-      when Time::Kind::Utc, Time::Kind::Unspecified
-        io << "+0000"
-      when Time::Kind::Local
-        negative, hours, minutes = local_time_zone_info
-        io << (negative ? "-" : "+")
-        io << "0" if hours < 10
-        io << hours
-        io << "0" if minutes < 10
-        io << minutes
+    def time_zone(with_seconds = false)
+      raise FloatingTimeConversionError.new if time.floating?
+
+      negative, hours, minutes, seconds = local_time_zone_info
+      io << (negative ? "-" : "+")
+      io << "0" if hours < 10
```

Comment:
These lines were not modified by this PR (only whitespace change), so I think this could be fixed independently.

Reply:
also char is slower than string. the people that thought that char is more efficient got it wrong. same when you index a string with a char instead of a string. or starts_with. char is slower. so string is preferred.

Reply:
@larubujo Sure about this? My benchmark says otherwise. However, this is completely off-topic here. So, if you have some evidence to support your claim, please post a new issue about that. This would be quite unexpected.

Reply:
@larubujo it's true indeed for `starts_with?(Char)` overload, its performance was severely  degraded since *v0.23.1*:

```cr
Benchmark.ips do |x|
  x.report("starts_with? Char") { "foo".starts_with? 'f' }
  x.report("starts_with? String") { "foo".starts_with? "f" }
end
```

*v0.23.1*:
```
  starts_with? Char  457.3M (  2.19ns) (±19.03%)  1.02× slower
starts_with? String 468.03M (  2.14ns) (±17.47%)       fastest
```

*master*:
```
  starts_with? Char  207.7M (  4.81ns) (±13.71%)  2.23× slower
starts_with? String 463.22M (  2.16ns) (±13.89%)       fastest
```

Yet, passing a `Char` to `IO` is still faster:

```cr
Benchmark.ips do |x|
  io = IO::Memory.new

  x.report("io << Char") { io << 'f' }
  x.report("io << String") { io << "f" }
end
```

*v0.23.1*:
```
  io << Char 139.61M (  7.16ns) (±19.08%)       fastest
io << String  82.44M ( 12.13ns) (±20.64%)  1.69× slower
```

*master*:
```
  io << Char 142.37M (  7.02ns) (±17.83%)       fastest
io << String  83.37M ( 11.99ns) (±13.60%)  1.71× slower
```

Reply:
it seems i was partially wrong. but it still slower for non-ascii:

```crystal
require "benchmark"

haystack = "がこう"

needle_string = "う"
needle_char = 'う'

a = 0
Benchmark.ips do |x|
  x.report("ends char") do
    a += haystack.ends_with?(needle_char) ? 1 : 0
  end
  x.report("ends string") do
    a += haystack.ends_with?(needle_string) ? 1 : 0
  end
end

Benchmark.ips do |x|
  x.report("starts char") do
    a += haystack.starts_with?(needle_char) ? 1 : 0
  end
  x.report("starts string") do
    a += haystack.ends_with?(needle_string) ? 1 : 0
  end
end

puts a
```

output

```
  ends char  95.17M ( 10.51ns) (±12.16%)  1.35× slower
ends string 128.26M (   7.8ns) (±17.20%)       fastest
  starts char  91.51M ( 10.93ns) (±11.07%)  1.45× slower
starts string 133.03M (  7.52ns) (±12.89%)       fastest
```

in any case times around nanosecond, so corrections like use '0' instead of "0" dont matter much

Reply:
I agree that it doesn't matter much, but if you consider the future evolution of the standard library then using the char overload gives you much more optimization possibilities, even if the current performance is different. 

If you notice a performance problem, make an issue please.